workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ScreenViewFactory.kt

```diff
@@ -93,3 +284,201 @@ public fun interface ViewStarter {
+ *        // Get the view factory of the wrapped screen.
+ *        wrapper.wrapped.toViewFactory(env)
+ *          // Transform it to factory that accepts OverrideNeutronFlow<W>, by
+ *          // replacing the OverrideNeutronFlow<W> with an EnvironmentScreen<W>
+ *          .unwrapping<OverrideNeutronFlow<W>, EnvironmentScreen<W>> {
+ *            it.wrapped.withEnvironment(
+ *              Environment.EMPTY + (NeutronFlowPolarity to it.polarity)
+ *            )
+ *          }
+ *          // Delegate to the transformed factory to build the view.
+ *          .buildView(wrapper, env, context, container)
+ *      }
+ *
+ * @param unwrap a function to extract [WrappedT] instances from [WrapperT]s.
+ */
+@WorkflowUiExperimentalApi
+public inline
+fun <reified WrapperT : Screen, WrappedT : Screen> ScreenViewFactory<WrappedT>.unwrapping(
+  crossinline unwrap: (wrapperScreen: WrapperT) -> WrappedT,
+): ScreenViewFactory<WrapperT> = unwrapping(unwrap) { _, ws, e, su -> su(unwrap(ws), e) }
+
+/**
+ * Transforms a [ScreenViewFactory] of [WrappedT] into one that can handle
+ * instances of [WrapperT].
+ *
+ * One general note: when creating a wrapper rendering, you're very likely to want it
+ * to implement [Compatible], to ensure that checks made to update or replace a view
+ * are based on the wrapped item. Each wrapper example below illustrates this.
+ *
+ * Also see the simpler variant of this function that takes only an [unwrap] argument.
+ *
+ * ## Example
+ *
+ * To make a wrapper that customizes [View] initialization:
+ *
+ *    class WithTutorialTips<W : Screen>(val wrapped: W) : Screen, Compatible {
+ *      override val compatibilityKey = Compatible.keyFor(wrapped)
+ *    }
+ *
+ *    fun <W: Screen> WithTutorialTipsFactory<W>() =
+ *      ScreenViewFactory.forBuiltView<WithTutorialTips<*>> = {
+ *        initialRendering, initialEnv, context, container ->
+ *          initialRendering.wrapped.toViewFactory(initialEnv)
+ *            .unwrapping<WithTutorialTips<W>, W>(
+ *              unwrap = { it.wrapped },
+ *              showWrapperScreen = { view, withTips, env, showUnwrapped ->
+ *                TutorialTipRunner.run(view)
+ *                showUnwrapped(withTips.wrapped, env)
+ *              }
+ *              .buildView(initialRendering, initialEnv, context, container)
+ *      }
+ *
+ * To make a wrapper that adds pre- or post-processing to [View] updates:
+ *
+ *    class BackButtonScreen<W : Screen>(
+ *       val wrapped: W,
+ *       val override: Boolean = false,
+ *       val onBackPressed: (() -> Unit)? = null
+ *    ) : Screen, Compatible {
+ *      override val compatibilityKey = Compatible.keyFor(wrapped)
+ *    }
+ *
+ *    fun <W: Screen> BackButtonViewFactory<W>() =
+ *      ScreenViewFactory.forBuiltView<BackButtonScreen<W>> {
+ *        initialRendering, initialEnv, context, container ->
+ *          initialRendering.wrapped.toViewFactory(initialEnv)
+ *            .unwrapping<BackButtonScreen<W>, W>(
```

Comment:
These four lines happen a lot. I wonder if we can do better.

```kotlin
fun <WrapperT, WrappedT> ScreenViewFactory.forScreenWrapper(
  unwrap: (WrapperT) -> WrappedT,
  mumbled: (Mumble, Mumble) -> Mumble.
)
```

Not in this PR.

Reply:
Tracked in #714.